iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp

```diff
@@ -9737,6 +9925,76 @@ TEST_F(RulesDrivenECPresentationManagerContentTests, LoadsPrimitiveArrayProperty
         << "Actual: \r\n" << BeRapidJsonUtilities::ToPrettyString(recordJson2["Values"]);
     }
 
+/*---------------------------------------------------------------------------------**//**
+* @bsitest
++---------------+---------------+---------------+---------------+---------------+------*/
+DEFINE_SCHEMA(LoadsGuidArrayPropertyValue, R"*(
+    <ECEntityClass typeName="MyClass">
+        <ECArrayProperty propertyName="GuidsArrayProperty" typeName="binary" extendedTypeName="BeGuid" />
+    </ECEntityClass>
+)*");
+TEST_F(RulesDrivenECPresentationManagerContentTests, LoadsGuidArrayPropertyValue)
+    {
+    // set up data set
+    ECClassCP ecClass = GetClass("MyClass");
+    BeGuid instanceGuid1;
+    instanceGuid1.FromString("182238d2-e836-4640-9b40-38be6ca49623");
+    BeGuid instanceGuid2;
+    instanceGuid2.FromString("814f3e14-63f2-4511-89a8-43ff3b527492");
+    RulesEngineTestHelpers::InsertInstance(s_project->GetECDb(), *ecClass, [instanceGuid1, instanceGuid2](IECInstanceR instance)
+        {
+        instance.AddArrayElements("GuidsArrayProperty", 2);
+        instance.SetValue("GuidsArrayProperty", ECValue((Byte*)&instanceGuid1, sizeof(BeGuid)), 0);
+        instance.SetValue("GuidsArrayProperty", ECValue((Byte*)&instanceGuid2, sizeof(BeGuid)), 1);
+        });
+
+    // create the rule set
+    PresentationRuleSetPtr rules = PresentationRuleSet::CreateInstance(BeTest::GetNameOfCurrentTest());
+    m_locater->AddRuleSet(*rules);
+
+    ContentRuleP rule = new ContentRule("", 1, false);
+    rules->AddPresentationRule(*rule);
+
+    ContentInstancesOfSpecificClassesSpecification* spec = new ContentInstancesOfSpecificClassesSpecification(1, "", ecClass->GetFullName(), false, false);
+    rule->AddSpecification(*spec);
+
+    // validate descriptor
+    ContentDescriptorCPtr descriptor = GetValidatedResponse(m_manager->GetContentDescriptor(AsyncContentDescriptorRequestParams::Create(s_project->GetECDb(), rules->GetRuleSetId(), RulesetVariables(), nullptr, 0, *KeySet::Create())));
+    ASSERT_TRUE(descriptor.IsValid());
+    ASSERT_EQ(1, descriptor->GetVisibleFields().size());
+    rapidjson::Document expectedFieldType;
+    expectedFieldType.Parse(Utf8PrintfString(R"({
+        "ValueFormat": "Array",
+        "TypeName": "binary[]",
+        "MemberType": {
+            "ValueFormat": "Primitive",
+            "TypeName": "binary"

```

Comment:
I wonder why extended type name wasn't carried over as `TypeName` like in `SetsTypeNameAsExtendedTypeNameIfProvided` test - can you check please?

Reply:
Fixed. Check `ECSchemaHelper::GetTypeName`